File: 2026/07-July/20-Monday/solution.java
class Solution {
    public List<List<Integer>> shiftGrid(int[][] grid, int k) {
        int rows = grid.length;
        int cols = grid[0].length;
        int total = rows * cols;

        k %= total;

        int[][] shifted = new int[rows][cols];

        for (int r = 0; r < rows; r++) {
            for (int c = 0; c < cols; c++) {

                int current = r * cols + c;
                int destination = (current + k) % total;

                int nr = destination / cols;
                int nc = destination % cols;

                shifted[nr][nc] = grid[r][c];
            }
        }

        List<List<Integer>> result = new ArrayList<>();

        for (int i = 0; i < rows; i++) {
            List<Integer> row = new ArrayList<>();

            for (int j = 0; j < cols; j++) {
                row.add(shifted[i][j]);
            }

            result.add(row);
        }

        return result;
    }
}

File: 2026/07-July/20-Monday/solution.js
/**
 * @param {number[][]} grid
 * @param {number} k
 * @return {number[][]}
 */
var shiftGrid = function (grid, k) {
    const rows = grid.length;
    const cols = grid[0].length;
    const total = rows * cols;

    k %= total;

    const shifted = Array.from(
        { length: rows },
        () => Array(cols).fill(0)
    );

    for (let r = 0; r < rows; r++) {
        for (let c = 0; c < cols; c++) {

            const current = r * cols + c;
            const target = (current + k) % total;

            const nr = Math.floor(target / cols);
            const nc = target % cols;

            shifted[nr][nc] = grid[r][c];
        }
    }

    return shifted;
};
